great_expectations/experimental/datasources/postgres_datasource.py

```diff
@@ -47,19 +47,63 @@ class ColumnSplitter:
     # values are the default values if a batch request using the splitter leaves the parameter unspecified.
     # template_params: List[str]
     # Union of List/Iterable for serialization
-    param_defaults: Dict[str, Union[List, Iterable]] = Field(default_factory=dict)
+    param_defaults: Dict[str, Union[List, Iterable]] = pydantic.Field(default_factory=dict)
 
     @property
     def param_names(self) -> List[str]:
         return list(self.param_defaults.keys())
 
 
+@pydantic_dc.dataclass(frozen=True)
+class BatchSorter:
+    metadata_key: str
+    reverse: bool = False
+
+
+BatchSortersDefinition: TypeAlias = Union[List[BatchSorter], List[str]]
+
+
+def _batch_sorter_from_str(sort_key: str) -> BatchSorter:
```

Comment:
Should this be a richer type? How will our users be aware of `-` or `+`? Documentation?

Reply:
The "rich" type is BatchSorter. This is a "quality of life" shorthand for people who want to use a string. The `+` and `-` are defined in the docstring. The rest of the string is arbitrary since it refers to a user defined column on their database table so I don't know what other checks to put in. 

This will also be used when reading the data asset definition in from config.